app/models/configuration_profile.rb

```diff
@@ -30,7 +31,7 @@ def activate!
   end
 
   def check_ongoing_mappings
-    return unless standards_organizations.any? {|dso| dso.mappings.any?(&:in_progress?) }
+    return unless mappings.any?(&:in_progress?)
```

Comment:
This will load all of the CP's mappings into memory, then loop through them looking for in-progress ones:

```sql
SELECT "mappings".* FROM "mappings" ...
```

It'd be better to use the `in_progress` scope and then (implicitly) count such mappings without actually loading anything:

```ruby
mappings.in_progress.any?  # will generate SELECT 1 AS one FROM "mappings" ...
```

Reply:
Good catch! Changing it.